crates/katana/executor/src/implementation/sir/mod.rs

```diff
@@ -71,7 +72,7 @@ impl ExecutorFactory for NativeExecutorFactory {
 pub struct StarknetVMProcessor<'a> {
     block_context: BlockContext,
     state: Arc<CachedState<StateProviderDb<'a>, PermanentContractClassCache>>,
-    transactions: Vec<(TxWithHash, Option<Receipt>)>,
+    transactions: Vec<(TxWithHash, Option<Receipt>, TxExecInfo)>,
```

Comment:
i was thinking, not sure if we should store the receipt and trace immediately here considering tx traces are pretty huge and can take some time for the conversion. not sure how this would affect tx execution tho.

maybe what we can do is we store it as `Box<dyn TransactionExecutionOutput>` and only take out the receipt and txinfo after the tx execution phase is done during block production process, instead of doing it after each tx execution.